### PR TITLE
Use bundle config to determine if BUNDLE_PATH must be set

### PR DESCRIPTION
**Pull request by milouse**
See: https://github.com/senny/rvm.el/issues/59

### CODE REVIEW

**Comment by milouse** [comment]
@zw963 and @akurashev can you take time to see if my changes fix your own problem? Thank you very much.

**Comment by akurashev** [comment]
@milouse I've checked and it works for me. Thank you!

**Comment by zw963** [comment]
Still not test it yet, but, why we hardcode "/home/milouse/" into RVM.el ?

thank you.

**Comment by milouse** [comment]
> Still not test it yet, but, why we hardcode "/home/milouse/" into RVM.el ?

It’s not in rvm.el itself, but in a separate unit test case. It’s just some random fake path used by the unit test framework.

**Comment by zw963** [comment]
@milouse , i tested it, it works for me too. thank you very much.

**Comment by milouse** [comment]
Thus, poke @sshaw I think we are all good here.

File: rvm.el [hunk 1]
```diff
@@ -344,6 +344,15 @@ function."
 (defun rvm--emacs-gempath ()
   (getenv "GEM_PATH"))
 
+(defun rvm--emacs-bundlepath ()
+  (let ((bundle-config (rvm--call-bundle-process "config get path"))
```
Comment (sshaw) [inline rvm.el:348]: The `get` option to `config` looks like it's only supported in Bundler 2.1. In older versions it'll set a config property named `get`. 

I think it makes sense to support older versions. Personally I've yet to work on a project using 2.1.

File: rvm.el [hunk 2]
```diff
@@ -344,6 +344,15 @@ function."
...
+        (search-start 0)
+        (bundle-paths '()))
+    (while (string-match ": \"?\\([^\"]*\\)\"?$" bundle-config search-start)
```
Comment (sshaw) [inline rvm.el:351]: Supporting older versions means looking for `"^path$"` then moving down 1 line attempting to match this regex. 

Would this be more accurate: `": \"\\([^\"]+\\)\"$"`? Don't we always want quotes with 1 or more chars between them or is there a reason to make it more lax?

File: rvm.el (hunk 1 above)
Comment (sshaw) [reply rvm.el:348]: [2.1](https://bundler.io/v2.1/man/bundle-config.1.html) and [2.0](https://bundler.io/v2.0/man/bundle-config.1.html) docs.

**Comment by sshaw** [comment]
Thanks @milouse. Checkout my comments and questions.

**Comment by milouse** [comment]
Le 17 octobre 2020 08:07:07 GMT+02:00, Skye Shaw <notifications@github.com> a écrit :
>Thanks @milouse. Checkout my comments and questions. 
>

Thank you very much for your review. I must admit I didn't think about older version. I'll install some older ruby/bundler and adapt it. Not before monday though.
-- 
Étienne Deparis

File: rvm.el (hunk 1 above)
Comment (milouse) [reply rvm.el]: You’re right. I’ve removed the get command, which is not required.

File: rvm.el (hunk 2 above)
Comment (milouse) [reply rvm.el]: No, there wasn’t any reason. I make it lax at the beginning of my tests as I was not very sure about the bundle config output format. However, now that I more confident about it, you’re right we can make it more strict.

**Comment by milouse** [comment]
Hi @sshaw, I’ve made the changes you suggested. They made me realize that my tests were not working as expected as I used the same name for the two test cases. It is now fixed.

File: rvm.el (hunk 2 above)
Comment (sshaw) [reply rvm.el]: > When I test old bundler version on my machine, the output format is always the same. Can you copy/paste an exemple of a different output you encounter?

Yes you're right. I forgot that one can do `bundle config path` with older versions. My comment assumed we'd have to parse the output of `bundle config`

**Comment by sshaw** [comment]
Lookin' good. Can you please squash into a single commit?

Do you notice an increase in time due to having to call `bundle`? No big deal, fix is great, but does seems noticeable to you?

File: rvm.el (hunk 2 above)
Comment (milouse) [reply rvm.el]: Oh, ok I understand. Thus no problem there :)

**Comment by milouse** [comment]
> Lookin' good. Can you please squash into a single commit?

Do as you prefer, I’ve no strong opinion on this, as long as the fix is deployed and appeal to everyone, I’ll be happy anyway :)

> Do you notice an increase in time due to having to call `bundle`? No big deal, fix is great, but does it seem noticeable to you?

I think not. In my own config, I’ve even add a hook every time I switch my buffer to select the right ruby version, and I don’t notice more hang than we can already have sometime while calling rvm… For me this change was transparent, but I’ve only relatively modern machine. Maybe it may induce a little freeze on some older machine? Anyway this bundle call should only occur when switching ruby environment. I don’t think it would be a real problem.

**Comment by sshaw** [comment]
> Lookin' good. Can you please squash into a single commit?

>> Do as you prefer...

I never knew that I could push to another users repo and branch by virtue of a PR being opened(?). 

Nevertheless rebased and merged! Thanks so much for swooping in with the fix @milouse! 🚀🤘

**Pull request merged**